src/basic/macro.h

```diff
@@ -213,7 +213,7 @@ static inline size_t GREEDY_ALLOC_ROUND_UP(size_t l) {
  *          Contrary to strlen(), this is a constant expression.
  * @x: a string literal.
  */
-#define STRLEN(x) (sizeof(""x"") - 1)
+#define STRLEN(x) ((unsigned) sizeof(""x"") - 1)
```

Comment:
I think it is probably a mistake tat CONST_MAX() checks for type compatibility like it currently does. It's not gcc that warns us here after all, it's the type compat check we ourselves added.

i.e. I think there's nothing wrong to use CONST_MAX() of two different sized integer types, as long as the signedness checks out

Reply:
https://github.com/systemd/systemd/pull/20346